universe/Building.h

```diff
@@ -214,9 +214,8 @@ class BuildingTypeManager {
 
 private:
     BuildingTypeManager();
-    ~BuildingTypeManager();
 
-    std::map<std::string, BuildingType*> m_building_types;
+    std::map<std::string, std::unique_ptr<BuildingType>> m_building_types;
```

Comment:
> To me returning... ```raw T*``` indicates nothing about ownership. 

I think the convention makes sense within (and only within) the 'no naked new' framework-- ***once you are confident your code employs no naked 'new's,*** and all heap allocations are required to be owned either via a unique_ptr or a shared_ptr, then by implication a raw pointer can never be used for ownership, and for improved efficiency you could allow this extra convention.  Here is another statement of what I think is essentially the same convention (or at least is an overlapping one): [Take smart pointers as parameters only to explicitly express lifetime semantics...A function that does not manipulate lifetime should take raw pointers or references instead.](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rr-smartptrparam)

Now, a quick grep of FreeOrion appears to show a great many naked 'new's still in our code, so perhaps it is premature to try jumping forward to this use of raw pointers.   Also, I still remember BigJoe's work going through replacing our uses of raw pointers with shared_ptr, and the idea of intentionally returning to using raw pointers in any situations evokes a bit of cognitive dissonance.

But overall this does seem like the right direction for us to be heading.